File: test/case01/go/main.go
package main

import (
	"context"
	"fmt"
	"os"
	"path/filepath"
	"runtime"
	"strings"

	agentlib "github.com/steven0lisa/agent-libs/go"
)

func main() {
	_, filename, _, ok := runtime.Caller(0)
	if !ok {
		fmt.Println("ERROR: cannot resolve current file path")
		return
	}

	repoRoot := filepath.Clean(filepath.Join(filepath.Dir(filename), "..", "..", ".."))
	workDir := filepath.Join(repoRoot, "test", "case01")

	promptPath := filepath.Join(workDir, "prompt.txt")
	promptBytes, err := os.ReadFile(promptPath)
	if err != nil {
		fmt.Printf("Error reading prompt: %v\n", err)
		return
	}
	prompt := string(promptBytes)

	fmt.Println("=== Prompt ===")
	fmt.Println(prompt)
	fmt.Println()

	dataPath := filepath.Join(workDir, "data.txt")
	dataBytes, err := os.ReadFile(dataPath)
	if err != nil {
		fmt.Printf("Error reading data: %v\n", err)
		return
	}
	data := string(dataBytes)
	fmt.Println("=== Data Preview ===")
	lines := strings.Split(data, "\n")
	for i := 0; i < 25 && i < len(lines); i++ {
		fmt.Println(lines[i])
	}
	fmt.Println("...")
	fmt.Println()

	apiKey := os.Getenv("ANTHROPIC_AUTH_TOKEN")
	if apiKey == "" {
		apiKey = os.Getenv("ANTHROPIC_API_KEY")
	}
	if apiKey == "" {
		fmt.Println("ERROR: ANTHROPIC_AUTH_TOKEN / ANTHROPIC_API_KEY not set")
		return
	}

	cfg := agentlib.DefaultConfig()
	cfg.APIKey = apiKey
	if baseURL := os.Getenv("ANTHROPIC_BASE_URL"); baseURL != "" {
		cfg.BaseURL = baseURL
	}
	if model := os.Getenv("ANTHROPIC_MODEL"); model != "" {
		cfg.Model = model
	}
	cfg.WorkDir = workDir
	cfg.MaxTurns = 10

	agent := agentlib.NewAgent(cfg)
	fmt.Println("=== Agent Output ===")
	fmt.Println()

	ctx := context.Background()
	events, err := agent.Run(ctx, prompt)
	if err != nil {
		fmt.Printf("Error starting agent: %v\n", err)
		return
	}

	for event := range events {
		switch event.Type {
		case agentlib.EventMessageDelta:
			fmt.Print(event.Data["text"])
		case agentlib.EventToolUseStart:
			fmt.Printf("\n[Tool call: %s] input=%v\n", event.Data["name"], event.Data["input"])
		case agentlib.EventComplete:
			fmt.Printf("\n\n=== Complete ===\n%v\n", event.Data["final_content"])
		case agentlib.EventError:
			fmt.Printf("\n[Error] %v\n", event.Data["message"])
		}
	}

	fmt.Printf("\n[Final state] %s\n", agent.State())
}

File: test/case01/rust/src/main.rs
use std::env;
use std::path::PathBuf;

use agentlib::{Agent, AgentConfig};

#[tokio::main]
async fn main() {
    let case_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
        .parent()
        .unwrap()
        .to_path_buf();

    let prompt_path = case_dir.join("prompt.txt");
    let prompt = tokio::fs::read_to_string(&prompt_path).await.unwrap();

    println!("=== Prompt ===");
    println!("{}", prompt);
    println!();

    let data_path = case_dir.join("data.txt");
    let data = tokio::fs::read_to_string(&data_path).await.unwrap();
    println!("=== Data Preview ===");
    for (i, line) in data.lines().enumerate() {
        if i >= 25 {
            println!("...");
            break;
        }
        println!("{}", line);
    }
    println!();

    let api_key = env::var("ANTHROPIC_AUTH_TOKEN")
        .or_else(|_| env::var("ANTHROPIC_API_KEY"))
        .unwrap_or_default();
    if api_key.is_empty() {
        println!("ERROR: ANTHROPIC_AUTH_TOKEN / ANTHROPIC_API_KEY not set");
        return;
    }

    let config = AgentConfig {
        api_key,
        base_url: env::var("ANTHROPIC_BASE_URL")
            .unwrap_or_else(|_| "https://api.anthropic.com".to_string()),
        model: env::var("ANTHROPIC_MODEL").unwrap_or_else(|_| "claude-sonnet-4-6".to_string()),
        work_dir: case_dir.clone(),
        max_turns: 10,
        ..Default::default()
    };

    let mut agent = Agent::new(config);
    println!("=== Agent Output ===\n");

    let mut rx = agent.run(&prompt).await.unwrap();

    while let Some(event) = rx.recv().await {
        match event {
            agentlib::Event::MessageDelta { text } => print!("{}", text),
            agentlib::Event::ToolUseStart { name, input, .. } => {
                println!("\n[Tool call: {}] input={}", name, input);
            }
            agentlib::Event::Complete { final_content } => {
                println!("\n\n=== Complete ===");
                println!("{}", final_content);
            }
            agentlib::Event::Error { message } => {
                println!("\n[Error] {}", message);
            }
            _ => {}
        }
    }

    println!("\n[Final state] {:?}", agent.state().await);
}
